eng/scripts/Save-Package-Namespaces-Property.ps1

```diff
@@ -0,0 +1,100 @@
+<#
+.SYNOPSIS
+Given a directory containing packageInfo json files and a repository root,
+add or update the namespaces. This script MUST be run in the build job after
+the build task.
+
+.DESCRIPTION
+Given a directory containing the packgeInfo json files, for each file call
+to get the namespaces from the DLL and populate the packageInfo file with
+the namespace information.
+
+.PARAMETER packageInfoDirectory
+The directory where the packageInfo files have been created. These files will
+be updated with namespaces information.
+
+.PARAMETER repoRoot
+The root of the repository. This, along with the information in the packageInfo
+files, is used to compute the the build path to the .dll so namespaces can be
+parsed from it. Using Azure.Template as an example, the path to the binary would
+be in $repoRoot/artifacts/bin/Azure.Template/Release/netstandard2.0/Azure.Template.dll
+#>
+[CmdletBinding()]
+Param (
+    [Parameter(Mandatory = $True)]
+    [string] $repoRoot,
+    [Parameter(Mandatory = $True)]
+    [string] $packageInfoDirectory
+)
+
+. (Join-Path $PSScriptRoot ".." common scripts common.ps1)
+Write-Host "repoRoot=$repoRoot"
+Write-Host "packageInfoDirectory=$packageInfoDirectory"
+if (-not (Test-Path -Path $packageInfoDirectory)) {
+    LogError "packageInfoDirectory '$packageInfoDirectory' does not exist."
+    exit 1
+}
+if (-not (Test-Path -Path $repoRoot)) {
+    LogError "repoRoot '$repoRoot' does not exist."
+    exit 1
+}
+# Get all of the packageInfo files
+$packageInfoFiles = Get-ChildItem -Path $packageInfoDirectory -Filter *.json
+if ($packageInfoFiles.Length -eq 0) {
+    LogWarning "packageInfoDirectory '$packageInfoDirectory' does not contain any json files. Please ensure that the packageInfo files were generated prior to running this."
+    exit 0
+}
+
+$foundError = $false
+# At this point the packageInfo files should have been already been created
+# and the only thing being done here is adding namespaces
+foreach ($packageInfoFile in $packageInfoFiles) {
+    Write-Host "processing $($packageInfoFile.FullName)"
+    $packageInfo = ConvertFrom-Json (Get-Content $packageInfoFile -Raw)
+    # Piece together the releaseFolderPath
+    $releaseFolderPath = Join-Path $repoRoot "artifacts" "bin" $packageInfo.Name "Release" "netstandard2.0"
+    Write-Host "releaseFolderPath=$releaseFolderPath"
+    if (Test-Path $releaseFolderPath) {
+        $defaultDll = Join-Path $releaseFolderPath "$($packageInfo.Name).dll"
+        if ($defaultDll -and (Test-Path $defaultDll)) {
```

Comment:
Could defaultDll ever be null here?

Reply:
No idea, but at this point I'm not sure anything would surprise me. Again, the check is cheap and I'd rather leave it here.